### PR TITLE
Java grader: replace command line arguments with parameters file

### PR DESCRIPTION
**Pull request by jonatanschroeder**
As discussed on Slack. Further details withheld from public discussion.

### CODE REVIEW

File: graders/java/autograder.sh [hunk 1]
```diff
@@ -42,20 +43,36 @@ fi
 
 RESULTS_TEMP_DIR=$(mktemp -d -p /grade/results)
 RESULTS_TEMP_FILE="$RESULTS_TEMP_DIR/$RANDOM.json"
```
Comment (nwalters512) [inline graders/java/autograder.sh:45]: TIL about `$RANDOM` 🤯

**Review by nwalters512 — APPROVED** [review]
It's been years since I've written any Java so I can't comment on that, but the overall strategy looks good to me.

File: graders/java/autograder.sh [hunk 2]
```diff
@@ -42,20 +43,36 @@ fi
...
+java -cp "$CLASSPATH" JUnitAutograder
 EOF
 
 if [ -f $RESULTS_TEMP_FILE ] ; then
-    mv $RESULTS_TEMP_FILE $RESULTS_FILE
+    RESULT_SIGNATURE=$(jq -r '.signature' $RESULTS_TEMP_FILE)
+    if [ "$RESULT_SIGNATURE" != "$SIGNATURE" ] ; then
+        echo Expected signature: $SIGNATURE
+        echo Provided signature: $RESULT_SIGNATURE
+        exception "Results did not contain correct signature. Exploit suspected. Please contact the instructor."
```
Comment (tdy) [inline graders/java/autograder.sh:72]: 😆 Should we deliberately make this more vague to encourage actual reporting? "Unable to process submission. Please contact the instructor."

Comment (echuber2) [reply graders/java/autograder.sh]: Along those lines, I'd recommend making the error message completely generic (like 0% grade, compilation error, or server error) with an explanation only appearing on the instructor log side.

Comment (echuber2) [reply graders/java/autograder.sh]: By generic I just mean indistinguishable from an error that would be emitted by some other code path. Indicating on the frontend whether the signature was the issue or not could be used for fuzzing or something, I imagine.

Comment (jonatanschroeder) [reply graders/java/autograder.sh]: Considering this is open source, I don't think it's that useful to make it too obscure, at the risk of instructors unable to understand why a student would get this error. I'm merging as is, future PRs can address that if necessary.

**Pull request merged**